nussl/core/audio_signal.py

```diff
@@ -1644,6 +1649,311 @@ def embed_audio(self, ext='.mp3', display=True):
         from . import play_utils
         return play_utils.embed_audio(self, ext=ext, display=display)
 
+    ##################################
+    #          Effect Hooks          #
+    ##################################
+
+    def build_effects(self, reset=True, overwrite=False):
+        """
+        Builds all effects in the AudioSignal's effects chains using the audio data of the calling 
+        AudioSignal. If reset=True, the current effects chains will be deleted. If overwrite=True, 
+        the calling AudioSignal's audio_data will be overwritten and the stft_data will be deleted. 
+
+        FFmpeg effects will be applied AFTER SoX effects. The effects time_stretch and pitch_shift 
+        are SoX effects. All others are FFmpeg effects.
+
+        If no effects have been added to the AudioSignal's effects chains, then build_effects 
+        will throw a RuntimeError. 
+
+        This will add a tremolo effect followed by a high pass effect to the AudioSignal's effects chains:
```

Comment:
I think you need a few more examples here. For instance, if you start a very easy one like  `time_stretch`, you can show the length of the signal to demonstrate how it should work. E.g.,

`>>> my_signal.duration`
`10.0`
`>>> my_signal.time_stretch(0.5)`
`>>> my_signal.build_effects(overwrite=True)`
`>>> my_signal.duration`
`5.0`

Then you can demonstrate daisy chaining and other advanced things.

Reply:
Added Example in docs